connection.go

```diff
@@ -32,10 +47,16 @@ func (c *legacyConnection) Connect(addr string, timeout time.Duration) error {
 		return err
 	}
 
-	c.Conn, err = net.DialTimeout("tcp", addr, timeout)
+	conn, err := net.DialTimeout("tcp", addr, timeout)
 	if err != nil {
 		return fmt.Errorf("legacy connection: dial: %w", err)
 	}
+
+	c.Conn = &writeTimeoutConn{
+		Conn:    conn,
+		timeout: timeout,
```

Comment:
This is using what the user believes is the connect timeout and not the read / write timeout, same for the ssh version.

Given the use is passing in a read / write / dial timeout the best option might be to just clarify it's use, thoughts?

Reply:
When using `NewClient()` with the option `Timeout()` the behavior didn't change. This is also documented for the `Timeout` option.  "Timeout sets read / write / dial timeout for a TeamSpeak 3 Client."

I added some documentation to the `Connect` methods for clarification though.